src/hotspot/share/classfile/classLoaderData.cpp

```diff
@@ -407,9 +407,6 @@ void ClassLoaderData::methods_do(void f(Method*)) {
 }
 
 void ClassLoaderData::loaded_classes_do(KlassClosure* klass_closure) {
-  // To call this, one must have the MultiArray_lock held, but the _klasses list still has lock free reads.
-  assert_locked_or_safepoint(MultiArray_lock);
-
```

Comment:
Do we no longer hold the lock or are you just missing the API to assert it is held?

Reply:
We no longer hold the lock in the callers to iterate through CLDG::_klasses list.  It was never needed.  CLDG::_klasses has iterators that are lock free (atomics)